### ISSUE TITLE
The implementation authenticates with provider not consumer

### ISSUE THREAD

**Issue by Andrew-Chen-Wang** [Andrew-Chen-Wang/django-social-provider-and-consumer-tutorial#4]
edit (took 5 minutes to figure out how this actually works... well hopefully) here is what I think needs to happen: https://github.com/Andrew-Chen-Wang/react-native-oauth-login/issues/1#issuecomment-860150418

**Comment by Andrew-Chen-Wang** [comment]
Maybe a helpful article explaining a little better!: https://www.toptal.com/django/integrate-oauth-2-into-django-drf-back-end

**Comment by Andrew-Chen-Wang** [comment]
btw transferred this issue from https://github.com/Andrew-Chen-Wang/react-native-oauth-login/issues for `mobile_consumer` implementation

**Comment by Andrew-Chen-Wang** [comment]
ok so that article from toptal has an image: https://www.toptal.com/django/integrate-oauth-2-into-django-drf-back-end

![](https://bs-uploads.toptal.io/blackfish-uploads/uploaded_file/file/193817/image-1582630068286-e1d8aa08f89edcdbd4f4cf3b4fd7884e.png)

basically saying that for mobile consumers. We first just authenticate with the provider then send the access/refresh tokens to the consumer. The consumer then gives the client a token pair for the consumer's endpoints where the payload (just extrapolating from my own knowledge here) encrypts the provider's access and refresh tokens.

This makes... less sense! Well it actually does make sense. We basically skip the entire allauth thing by only using allauth's models and adapters to save stuff. The flow is the same though; we just need to create an EmailAddress model, save some stuff in AllAuth's social account models, and continue on!

**Comment by Andrew-Chen-Wang** [comment]
Also the reason we see some random username is because of this: https://github.com/pennersr/django-allauth/blob/c50a30acc0e4553297b0ddb697d55a290ed6d038/allauth/account/adapter.py#L196-L218

**Comment by Andrew-Chen-Wang** [comment]
Completed: 0fd63bff7386c60a2f95eaea5dba21ce01a89b32

Took one evening to realize that the redirect scheme actually needed to be set in the provider lmao... but thank goodness I saw that article about the token exchange. Nothing else would've convinced me that was "safer" tbh. I did think of it but I was thinking too complex and though NAAA exchanging tokens willy nilly? Back then, it riddled down to a debate: should the consumer have my access tokens? And the answer is the web consumer already does it, I just didn't know lmao
